### PR TITLE
add content safety internal service

### PR DESCRIPTION
**Pull request by suxi-ms**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/cognitiveservices/ContentSafetyInternal/models.tsp [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+enum ContentSafetyOperation {
+  AnalyzeText,
+  AnalyzeImage,
+  @added(ContentSafetyInternal.Versions.v2023_05_30_Preview)
+  @removed(ContentSafetyInternal.Versions.v2023_10_01)
+  AnalyzeImageWithText
+}
+
+@doc("Content safety API version.")
+enum ContentSafetyVersion {
```
Comment (msyyc) [inline specification/cognitiveservices/ContentSafetyInternal/models.tsp:57]: As SDK owner, we advise to make enum name for each value like: 
```
@doc("Content safety API version.")
enum ContentSafetyVersion {
  v2023_04_30_Preview: "2023-04-30-preview",
  @added(ContentSafetyInternal.Versions.v2023_05_30_Preview)
  @removed(ContentSafetyInternal.Versions.v2023_10_01)
  v2023_05_30_Preview: "2023-05-30-preview",
  @added(ContentSafetyInternal.Versions.v2023_10_01)
  v2023_10_01: "2023-10-01"
}
```

File: specification/cognitiveservices/ContentSafetyInternal/tspconfig.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  #   package-dir: "contentsafety-rest"
+  #   generateMetadata: true
+  #   generateTest: true
+  #   azureSdkForJs: true
+  #   packageDetails:
+  #     name: "@azure-rest/contentsafety-rest"
+  #     description: "ContentSafety Service"
+  # "@azure-tools/typespec-java":
+  #   package-dir: "contentsafety"
+  #   namespace: com.contentsafety
```
Comment (msyyc) [inline specification/cognitiveservices/ContentSafetyInternal/tspconfig.yaml:31]: It is better to open Js/Java SDK generaion.

File: specification/cognitiveservices/ContentSafetyInternal/routes.tsp [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  extends Azure.Core.ResourceOperations<NoRepeatableRequests &
+      NoConditionalRequests &
+      NoClientRequestId> {}
+
+
+interface WhitelistFeature {
+  @route("whitelist/features")
+  @summary("List Whitelist Features.")
+  @doc("Get whitelist features.")
+  op listBlocklistTopTerms is RpcOperation<
```
Comment (weidongxu-microsoft) [inline specification/cognitiveservices/ContentSafetyInternal/routes.tsp:29]: The naming of op seems not correct. `listWritelistFeatures`?

File: specification/cognitiveservices/ContentSafetyInternal/tspconfig.yaml (hunk 2 above)
Comment (weidongxu-microsoft) [reply specification/cognitiveservices/ContentSafetyInternal/tspconfig.yaml:31]: Java PR https://github.com/Azure/azure-rest-api-specs/pull/25307

File: specification/cognitiveservices/ContentSafetyInternal/routes.tsp (hunk 3 above)
Comment (suxi-ms) [reply specification/cognitiveservices/ContentSafetyInternal/routes.tsp:29]: Thanks for review. This a PR for internal review. I will abondon the PR to reduce your work.

File: specification/cognitiveservices/ContentSafetyInternal/models.tsp (hunk 1 above)
Comment (suxi-ms) [reply specification/cognitiveservices/ContentSafetyInternal/models.tsp:57]: Thanks for review. This a PR for internal review. I will abondon the PR to reduce your work.

**Pull request closed without merging**